agent/bzero/registration/registration.go

```diff
@@ -121,13 +115,31 @@ func post(log logger.T, regInfo BZeroRegRequest, regUrl string) (*http.Response,
 
 	// Keep looping through our ticker, waiting for it to tell us when to retry
 	for range ticker.C {
-		// Make our Client
+
+		// Make our http Client
 		var httpClient = &http.Client{
 			Timeout: time.Second * 10,
 		}
 
+		// Build Registration Endpoint
+		if serviceUrl == "" {
+			serviceUrl = prodServiceUrl
+		}
+		regUrl, err := url.Parse(serviceUrl)
+		if err != nil {
+			return response, fmt.Errorf("could not parse service url: %s error: %s", serviceUrl, err)
+		}
+
+		// Get connection service url from bastion
+		connectionServiceUrl, connectionServiceUrlErr := getConnectionServiceUrlFromServiceUrl(log, httpClient, serviceUrl)
+		if connectionServiceUrlErr != nil {
+			return &http.Response{}, connectionServiceUrlErr
+		}
```

Comment:
Retry logic is lost if we return like this. Maybe it's not a big deal, but it's technically an inconsistency between what is live right now.

We need something like this:

```
f response.StatusCode == http.StatusInternalServerError ||
			response.StatusCode == http.StatusBadRequest ||
			response.StatusCode == http.StatusNotFound ||
			response.StatusCode == http.StatusUnauthorized ||
			response.StatusCode == http.StatusUnsupportedMediaType {
			ticker.Stop()
			log.Infof("Registration Endpoint: %s", regUrl)
			log.Infof("Registration Request Body: %s", string(regInfoBytes))
			return response, fmt.Errorf("received response code: %d, not retrying", response.StatusCode)
		}
		if err != nil || response.StatusCode != http.StatusOK {
			continue
}
```

so maybe pull that logic into function so we can reuse it for the get conn service url endpoint as well

Reply:
Added retry as more general function in 77a2049